apps/kg/dataloader/KGDataset.py

```diff
@@ -463,77 +498,72 @@ def get_dataset(data_path, data_name, format_str, files=None):
     return dataset
 
 
-def get_partition_dataset(data_path, data_name, format_str, part_id):
-    part_name = os.path.join(data_name, 'part_'+str(part_id))
-    if format_str == 'built_in':
-        if data_name == 'Freebase':
-            dataset = KGDatasetFreebase(data_path, part_name, read_triple=True, only_train=True)
-        elif data_name == 'FB15k':
-            dataset = KGDatasetFB15k(data_path, part_name, read_triple=True, only_train=True)
-        elif data_name == 'FB15k-237':
-            dataset = KGDatasetFB15k237(data_path, part_name, read_triple=True, only_train=True)
-        elif data_name == 'wn18':
-            dataset = KGDatasetWN18(data_path, part_name, read_triple=True, only_train=True)
-        elif data_name == 'wn18rr':
-            dataset = KGDatasetWN18rr(data_path, part_name, read_triple=True, only_train=True)
-        else:
-            assert False, "Unknown dataset {}".format(data_name)
-    elif format_str == 'raw_udd':
-        # user defined dataset
-        assert False, "When using partitioned dataset, we assume dataset will not be raw"
-    elif format_str == 'udd':
-        # user defined dataset
-        format = format_str[4:]
-        dataset = KGDatasetUDD(data_path, data_name, files, format, read_triple=True, only_train=True)
+def get_partition_dataset(data_path, data_name, part_id):
+    part_name = os.path.join(data_name, 'partition_'+str(part_id))
+    path = os.path.join(data_path, part_name)
+
+    if not os.path.exists(path):
+        print('Partition file not found.')
+        exit()
+
+    train_path = os.path.join(path, 'train.txt')
+    local2global_path = os.path.join(path, 'local_to_global.txt')
+    partition_book_path = os.path.join(path, 'partition_book.txt')
+
+    if data_name == 'Freebase':
+        relation_path = os.path.join(path, 'relation2id.txt')
+        skip_first_line = True
     else:
-        assert False, "Unknown format {}".format(format_str)
+        relation_path = os.path.join(path, 'relations.dict')
+        skip_first_line = False
```

Comment:
what does it mean? does it support other datasets? does it support raw_udd and udd?